### DISCUSSION TITLE
What are the current usage control capabilities of the EDC?

### DISCUSSION THREAD

**Discussion by soniasantiago** [eclipse-edc/Connector#447] [Q&A] [answered]
Hi, 
I'm currently collaborating in a project in which we use the IDS Dataspace Connector (DSC) implemented by Fraunhofer. However, since development of this connector is being interrupted to focus on the EDC, we have started having a look to the implementation of the EDC.

Among other topics, we are interested in the definition and verification at execution time of usage control policies. For example, that a dataset can only be requested to a provider connector withing a certain interval, or a certain number of times withing an interval of time. The DSC implements some of this usage control policies (see [this link](https://international-data-spaces-association.github.io/DataspaceConnector/Documentation/v6/UsageControl)). 

Thus, my question is whether the EDC implements some kind of usage control policy. If so, could you please give me some information about that? Moreover, if there is some documentation on this, I would appreciate if you could you provide any link to it.

I have searched a little bit on the source code and found some classes under packages:
* org.eclipse.dataspaceconnector.policy.engine;
* package org.eclipse.dataspaceconnector.policy.model;

that might be related to this topic, in the sense that they can be used as a base to do the implementation for the definition and validation of usage control policies, but I'm unsure of this. 

Many thanks in advance.
Best regards, 

Sonia.

**Comment by juliapampus** [comment, marked as answer]
Hi @soniasantiago! You can find the current policy documentation [here](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/docs/Policies.md). This will be extended soon. The EDC uses ODRL as its internal policy language but is also able to map this to IDS during IDS communication. Currently, we are about to define policy classes we want to support and implement. Therfore, I unfortunately cannot give you further information about providing the same policy functionality as the DSC, apart from telling you that this is what we're aiming for. We'll keep you up to date.

**Reply by soniasantiago** [reply]
Hi @juliapampus!
Many thanks for the information and the link. I have attended to some recent talks on the EDC where this topic was also discussed, and I could see that this topic is in the roadmap, so I will be keeping an eye on it, too.
Regards,

Sonia.
